### ISSUE TITLE
Popover id causes broken ARIA reference

### ISSUE THREAD

**Issue by thoughtbyte** [react-bootstrap/react-bootstrap#1827]
Hi,

The documentation for the Popover component states that the id prop is required for accessibility. But I think it actually causes an error. The problem is that when given an id, the child element (e.g. the button that triggers the popover) is given an aria-describedby attribute with that id. However, the Popover div that is assigned this id does not actually exist until the OverlayTrigger is triggered (i.e. the button is clicked).

This causes the error "Broken ARIA reference" when scanned with WebAIM's WAVE toolbar extension. WebAIM is one of the leaders in accessibility and are referenced in react-bootstrap's own roadmap, so it seems like this is an error that should be considered.

From the perspective of someone using a screen reader, if there are a few Popovers on a site, then there will be a few aria-describedbys that refer to something that doesn't exist yet. This will likely cause confusion.

Unfortunately I haven't been able to come up with a solution, but I look forward to discussing the problem. Right now I just don't use ids in my Popovers.

**Comment by vga-** [comment]
I just did some digging around Bootstrap itself and it appears that bootstrap avoids these broken ARIA references by adding the aria-described by attribute only when the trigger has been triggered. This means that when the popover is not displayed, there is no aria-described by. When the trigger is triggered, the aria-described by of the trigger is set, and the popover div appears. Hence, there is never a point at which there is a broken reference.

Note: this applies to tooltips as well.

**Comment by vga-** [comment]
I think I've found a solution to the problem, if it is a problem, in setting the `aria-describedby` prop conditionally in the OverlayTrigger depending on whether `isOverlayShown` is true or false. This will bring the component in line with the behavior of the vanilla Bootstrap overlays.

I will submit a pull request tonight, but I am interested in others' thoughts.

**Comment by taion** [comment]
Sounds good. Looking forward to it.

**Comment by jquense** [comment]
I would like to get some feedback from folks with more experience with screen Readers. My personal experience is that they don't really appreciate elements to loose and gain some `aria` dynamically. My concern here would be that when the aria-describedby is added the SR won't notice and announce it.

**Comment by vga-** [comment]
I will add that the broken reference error is a pretty big deal. So it might be the lesser of two evils to remove and add the aria-described by label dynamically (this is the Bootstrap route) even if screen reading dynamic arias is a problem.

This concern came up because the app we're building has to be as close to WebAIM's guidelines--we do pretty extensive testing with screen readers. I'll test this proposed solution out and see what happens and report back. Thanks for adding the concern!

**Comment by vga-** [comment]
An example that MDN links to: http://www.oaa-accessibility.org/examplep/tooltip1/

This example takes another route and seems to allay both concerns.

Rather than adding and removing the popover (or tooltip) DOM element, it `display: inline`s and `display: none`s it. The `aria-describedby` attribute is then always present, but the `aria-hidden={boolean}` attribute of the tooltip or popover changes dynamically when then the trigger is triggered.

This solution seems to solve both issues, but it also seems like a much bigger change to the current behavior of popovers. I have no idea how feasible this kind of change is, and what kinds of consequences it would have performance wise, etc.
